### DISCUSSION TITLE
Bald Earth less than 0 elevation

### DISCUSSION THREAD

**Discussion by miles-tbe** [pelicanmapping/osgearth#1993] [Support Q&A] [answered]
I have an earth file I am using where I only define an image for the globe and no elevation. When I use the following function:
```cpp
MapNodePtr->getTerrain()->getHeight (MapNodePtr->getMapSRS()->getGeographicSRS(), InputLon, InputLat, &TerrainHeight);
```
I only get values back in TerrainHeight are below 0. I would expect no elevation means 0 elevation. Is there a way to be able to have an earth that would be visually and numerically "flat"?  

Thanks,
Miles

**Comment by gwaldron** [comment, marked as answer]
The `getTerrain()` API is an interface to the in-memory scene graph representation the the terrain. In other words, a triangle mesh approximating the curve of the earth. So the result is correct. If you want to instead poll the elevation data itself you can use the elevation pool at `MapNode::getElevationPool()` and call `getSample`.

**Reply by miles-tbe** [reply]
> 

This is definitely more what I was looking for. I have one follow up question and if I can get an answer to that I will be good on this issue. This works good for when I have terrain everywhere but when I have an earth file with no elevation it obviously can't get a pool. Is there a way in osgEarth to find out whether there is terrain data to read or not? As in, something I can put inside an if statement to control when to set the terrain and when I need to override the lack of terrain to 0? 

Thanks

**Reply by miles-tbe** [reply]
I found the ElevationSample Sample.hasData() function that solves my issue.
